kaolin/graphics/Lighting.py

```diff
@@ -255,12 +264,10 @@ def compute_directional_light(
     if directional_intensity == 0:
         return light
 
-    # Unwrap the batchsize dimension (for easier vectorization).
-    faces = face_vertices.reshape(batchsize * num_faces, 3, 3)
     # Compute face normals.
```

Comment:
https://github.com/NVIDIAGameWorks/kaolin/blob/master/kaolin/rep/TriangleMesh.py#L154
https://github.com/NVIDIAGameWorks/kaolin/blob/master/kaolin/models/meshcnn.py#L54

I don't understand why we won't use one of those

Reply:
This convention is tweaked (as opposed to using our original), for exactness wrt softras. (In fact our first release used our function from `TriangleMesh`)

Reply:
Yes, but it's exactly the same than the one for MeshCNN then, right ?

Here is a suggestion, let add an option to https://github.com/NVIDIAGameWorks/kaolin/blob/master/kaolin/rep/TriangleMesh.py#L154
like `robust` or `averaged` (or whatever argument name that would indicate that you average over all cross products instead of picking just one).

It would else for reusability and would avoid out-of-the-core calculation on MeshCNN.

Reply:
Yeah, I think that'll be great. We'd want these to be "inside core" and standardized to the extent possible